shell/platform/android/platform_view_android_jni_impl.cc

```diff
@@ -1094,20 +1136,57 @@ void PlatformViewAndroidJNIImpl::SurfaceTextureDetachFromGLContext(
   FML_CHECK(CheckException(env));
 }
 
-void PlatformViewAndroidJNIImpl::FlutterViewOnDisplayPlatformView(int view_id,
-                                                                  int x,
-                                                                  int y,
-                                                                  int width,
-                                                                  int height) {
+void PlatformViewAndroidJNIImpl::FlutterViewOnDisplayPlatformView(
+    int view_id,
+    int x,
+    int y,
+    int width,
+    int height,
+    MutatorsStack mutators_stack) {
   JNIEnv* env = fml::jni::AttachCurrentThread();
-
   auto java_object = java_object_.get(env);
   if (java_object.is_null()) {
     return;
   }
 
+  jobject mutatorsStack = env->NewObject(g_mutators_stack_class->obj(),
+                                         g_mutators_stack_init_method);
+
+  std::vector<std::shared_ptr<Mutator>>::const_iterator iter =
+      mutators_stack.Begin();
+  while (iter != mutators_stack.End()) {
```

Comment:
do you know how often this line https://github.com/flutter/engine/blob/712f619737f1323fd2908b397aa8bd64fb2ccbe5/shell/platform/android/external_view_embedder/external_view_embedder.cc#L39  will exit? I wonder if by doing this work earlier we save some cycles.

Reply:
this exits when there is no change to the platform view layout. it is quite often.
We definitely want to optimize it to only call `onDisplayPlatformView` when there are changes in the EmbeddedViewParams. Let's do it in a separate PR?

"I wonder if by doing this work earlier we save some cycles."
What do you mean by this? Doing which work earlier?

Reply:
> What do you mean by this? 

Mapping the mutator stack to a Java object could be done earlier to save cycles, since it would be done only when there are changes to the platform view layout. That would require wiring up another JNI method.  This can be done in a separate PR though.

Reply:
SG